### ISSUE TITLE
[GR-57252] Improve error message when trying to build static image on unsupported platforms

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#9490]
Currently, the error message on Linux AArch64 (on which static linking is not supported) looks like this:

```
[1/8] Initializing... (0.0s @ 0.08GB)
Error: Building images for org.graalvm.nativeimage.Platform$LINUX_AARCH64 (target libc: musl) requires static JDK libraries.
Use most recent JDK from https://github.com/graalvm/labs-openjdk-*/releases
Missing libraries: java, nio, net, zip
(search path: /home/github/actions-runner/_work/_tool/graalvm-jdk-17_linux-aarch64_bin/17.0.0/arm64/graalvm-jdk-17.0.12+8.1/lib/static/linux-aarch64/musl) 
```

The reference to LabsJDK can be confusing to users as it may suggest you can build on the platform with a different JDK. Instead, the error message should clearly state that static linking is not supported on this platform.

**Comment by mthmulders** [comment]
@fniephaus, pending a better error message as this issue describes, are there any plans to eventually enable static linking on linux/aarch64?

**Comment by fniephaus** [comment]
Yes, we have plans to add support for static linking on linux/aarch64. Not in JDK 25, but hopefully soon.

**Comment by codingkiddo** [comment]
Hi @fniephaus / @wirthi, I’d like to help with this issue if it is still open for contributions.

From the discussion, I understand that full static linking support for Linux AArch64 is planned separately, while this issue is specifically about improving the current Native Image error message.

Would a small PR be useful that keeps the scope limited to:

1. detecting the unsupported Linux AArch64 + musl/static-linking case earlier or more precisely,
2. replacing the misleading LabsJDK suggestion with an actionable “static linking is not supported on this platform” message, and
3. adding regression coverage for the error message?

I’ll avoid changing platform support behavior and keep the PR focused only on diagnostics/test coverage.

**Comment by wirthi** [comment]
Hey @codingkiddo 

sure, if you have a concise PR that can improve the current situation and improve the error message, we are happy to have a look and adopt it if possible. Note that you will not have the OCA signed so we can adopt any code.

Christian
